src/librustc_typeck/astconv.rs

```diff
@@ -483,8 +484,23 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             arg.descr(),
             kind,
         );
+
+        let kind_ord = match kind {
+            "lifetime" => ParamKindOrd::Lifetime,
+            "type" => ParamKindOrd::Type,
+            "constant" => ParamKindOrd::Const,
+            _ => panic!(),
```

Comment:
Is the right way to trigger an ICE?

Reply:
I believe that `bug!` is the preferred way to trigger an ICE.

In this case, I would recommend changing `opt_span_bug_fmt` to take the original `GenericParamDefKind`, rather than its string description. That will allow you to write an exhaustive match, removing the need for an explicit `panic!`/`bug!`

Reply:
Thanks for the tip! I'm new to contributing to rustc, so could you describe the change you're recommending in more detail?

Reply:
On closer examination, it turns out that the two callers of this function actually have two different "original" types: `GenericParamDefKind` and `GenericArg`.

I believe your change is actually perfectly correct.

Reply:
I went through the same thought process @Aaron1011. It would be a little cleaner to pass `ParamKindOrd` directly, but I'm not sure it's worth making more changes.

Reply:
It would be easy to change in this pull request if desired, but I'm not sure it's worth doing: it adds quite a bit of boilerplate code, and it's only used in this one case. If we ever find ourselves doing something similar again in the future, we should revisit it.